### PR TITLE
Add user auth

### PR DESCRIPTION
**Pull request by jjgonecrypto**
adding simple user authentication from devise gem.

### CODE REVIEW

**Comment by jjgonecrypto** [comment]
also removing redundant scaffolding around schemes (which are now embedded within services)

File: app/models/user.rb [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+  #field :name
+  
+  validates_presence_of :username
+  validates_uniqueness_of :username, :email, :case_sensitive => false
+  attr_accessible :username, :email, :password, :password_confirmation, :remember_me
   
   has_many :subscriptions, :dependent => :delete
   
-  attr_protected :password
+  #attr_protected :passworddd
```
Comment (jonstorer) [inline app/models/user.rb]: Do you need all those d's?

File: spec/models/subscription_spec.rb [hunk 2]
```diff
@@ -78,7 +78,7 @@
   context "ensure subscription mailer" do
     let(:bill) { Date.today }
     let!(:user1) { Factory(:user) }
-    let!(:user2) { Factory(:user) }
+    let!(:user2) { Factory(:user, username: user1.username + '1', email: '1' + user1.email) }
```
Comment (jonstorer) [inline spec/models/subscription_spec.rb]: Why the +1

Comment (jjgonecrypto) [inline spec/models/subscription_spec.rb]: because the user model now has unique username/email validation checks. is there a better way to do that?

Comment (jonstorer) [inline spec/models/subscription_spec.rb]:
``` ruby
Factory.sequence :email do |n|
  "user#{n}@example.com"
end
```

``` ruby
Factory.define :user do |user|
  user.email  { Factory.next :email }
end
```

File: spec/factories/users.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 Factory.define :user do |user|
   user.username "username"
+  user.email "something@something.xxx"
```
Comment (jonstorer) [inline spec/factories/users.rb]: `something.xxx` could be real... and `example.com` and `example.org` are reserved for documentation and testing.  http://www.iana.org/domains/example/

File: spec/models/subscription_spec.rb (hunk 2 above)
Comment (jjgonecrypto) [inline spec/models/subscription_spec.rb]: done

**Pull request merged**